module_utils/network_lsr/argument_validator.py

```diff
@@ -1802,3 +1802,22 @@ def validate_connection_one(self, mode, connections, idx):
                     idx,
                     "ip.ipv6_disabled is not supported by initscripts.",
                 )
+
+        if connection["ip"]["dns_options"] or connection["ip"]["dns_search"]:
+            addrs4 = list(
+                [
+                    address
+                    for address in connection["ip"]["address"]
+                    if address["family"] == socket.AF_INET
+                ]
+            )
+            if not connection["ip"]["dhcp4"] and not addrs4:
+                raise ValidationError.from_connection(
+                    idx,
+                    "IPv4 needs to be enabled to support IPv4 nameservers.",
+                )
+            if connection["ip"]["ipv6_disabled"]:
+                raise ValidationError.from_connection(
+                    idx,
+                    "IPv6 needs to be enabled to support IPv6 nameservers.",
+                )
```

Comment:
@thom311 , in network role, there is no explicit option like `ipv4_disabled`, how do you like the validation 
```
            if not connection["ip"]["dhcp4"] and not addrs4:
                raise ValidationError.from_connection(
                    idx,
                    "IPv4 needs to be enabled to support IPv4 nameservers.",
                )
 ```